public/getabook.html

```diff
@@ -193,33 +196,89 @@ <h2>Get a book </h2>
 
   }
 
-  donationForm.addEventListener('submit',e=>{
-    if(validate_phone(document.getElementById('contact'))){
-    e.preventDefault();
-    const { data, error } = _supabase
-      .from('getabook_form')
-      .insert([
-        {
-          email:document.getElementById('email').value,
-         
-          contact : document.getElementById('contact').value,
-          zip : document.getElementById('zip').value,
-        }
-      ]).then(res=>{
-        console.log(res);
-        if(res.error){
-          alert(res.error.message);
-        }
-        else{
-        alert('Data Added');
+  const email = document.getElementById('email');
+  const contact = document.getElementById('contact');
+  const zip = document.getElementById('zip');
+
+  function validateIssue(){
+    console.log("called");
+    if(checkInputs())
+    return true;
+    else
+    return false;
+
+  }
+
+  function checkInputs()
+  {
+       //get values from input fields
+       const emailval = email.value.trim();
+       const contactval = contact.value.trim();
+        const zipval = zip.value.trim();
+     
+      
+       if(emailval === '')
+       { setErrorFor(email,'Please enter a E-mail');}
+       else if(!validEmail(emailval))
+       { setErrorFor(email,'Please enter a valid E-mail');}
+        else if(contactval == ""){
+          setSuccessFor(email);
+        setErrorFor(contact,"Please enter a contact no");
+        }else if(contactval.length!=10)
+        { setErrorFor(contact,'Please enter a valid 10 digit contact no');}
```

Comment:
Since you changed the type of input to `text`, and there seems to be no check if the `contactval` contains all numbers.
A random 10 character string will also pass this. So this should be changed..
Plus I think we should keep it to `tel` only....